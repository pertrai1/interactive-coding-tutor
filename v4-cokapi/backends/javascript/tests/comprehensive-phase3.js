// Comprehensive test showing all Phase 3 features working together

// Module imports (transformed to mock system)
import { add } from './math';

// Advanced object features
const name = "JavaScript";
const config = {
  name,                               // Object shorthand
  getValue() {                        // Method definition
    return this.name;
  }
};

// Enhanced array methods with chaining
const numbers = [1, 2, 3, 4, 5];
const result = numbers
  .filter(n => n % 2 === 0)          // Filter even numbers
  .find(n => n > 2);                 // Find first > 2

// Test includes method
const hasThree = numbers.includes(3);

// Using imported function (mock module system)
const sum = add(5, 7);