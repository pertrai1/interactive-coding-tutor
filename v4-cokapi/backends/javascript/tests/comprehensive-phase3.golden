{"code":"// Comprehensive test showing all Phase 3 features working together\n\n// Module imports (transformed to mock system)\nimport { add } from './math';\n\n// Advanced object features\nconst name = \"JavaScript\";\nconst config = {\n  name,                               // Object shorthand\n  getValue() {                        // Method definition\n    return this.name;\n  }\n};\n\n// Enhanced array methods with chaining\nconst numbers = [1, 2, 3, 4, 5];\nconst result = numbers\n  .filter(n => n % 2 === 0)          // Filter even numbers\n  .find(n => n > 2);                 // Find first > 2\n\n// Test includes method\nconst hasThree = numbers.includes(3);\n\n// Using imported function (mock module system)\nconst sum = add(5, 7);","trace":[{"line":6,"event":"step_line","func_name":"<module>","globals":{},"ordered_globals":[],"stack_to_render":[],"heap":{},"stdout":""},{"line":7,"event":"step_line","func_name":"<module>","globals":{},"ordered_globals":[],"stack_to_render":[],"heap":{},"stdout":""},{"line":9,"event":"step_line","func_name":"<module>","globals":{},"ordered_globals":[],"stack_to_render":[],"heap":{},"stdout":""},{"line":12,"event":"step_line","func_name":"<module>","globals":{"_math":["INSTANCE","Object"]},"ordered_globals":["_math"],"stack_to_render":[],"heap":{},"stdout":""},{"line":15,"event":"step_line","func_name":"<module>","globals":{"_math":["INSTANCE","Object"],"name":"JavaScript"},"ordered_globals":["_math","name"],"stack_to_render":[],"heap":{},"stdout":""},{"line":21,"event":"step_line","func_name":"<module>","globals":{"_math":["INSTANCE","Object"],"name":"JavaScript","config":["INSTANCE","Object"]},"ordered_globals":["_math","name","config"],"stack_to_render":[],"heap":{},"stdout":""},{"line":30,"event":"step_line","func_name":"<module>","globals":{"_math":["INSTANCE","Object"],"name":"JavaScript","config":["INSTANCE","Object"],"numbers":["LIST",1,2,3,4,5],"result":4,"hasThree":true},"ordered_globals":["_math","name","config","numbers","result","hasThree"],"stack_to_render":[],"heap":{},"stdout":""},{"line":33,"event":"step_line","func_name":"<module>","globals":{"_math":["INSTANCE","Object"],"name":"JavaScript","config":["INSTANCE","Object"],"numbers":["LIST",1,2,3,4,5],"result":4,"hasThree":true,"sum":12},"ordered_globals":["_math","name","config","numbers","result","hasThree","sum"],"stack_to_render":[],"heap":{},"stdout":""}]}
